content/developers/developer-patterns/navigating-merklelogs/index.md

```diff
@@ -0,0 +1,501 @@
+h = hashlib.sha256()
+h.update(bytes([0]))
+h.update(bytes.fromhex("018e84dbbb6513a6"[2:]))
+h.update("assets/31de2eb6-de4f-4e5a-9635-38f7cd5a0fc8/events/21d55b73-b4bc-4098-baf7-336ddee4f2f2".encode())
+h.hexdigest()
+```
+
+The variable portion *for the first massif* contains exactly *16383* MMR *nodes*.
+Of those nodes, *8192* are the leaf entries in the Merkle tree corresponding to your events.
+
+When a massif is initialized the trieData is pre-populated for all leaves and set to all zero bytes.
+As events are recorded in the log, the zero-padded index is filled in.
+A sub-range of field 0 will change when saving the last idtimestamp in it.
+The mmr node values are strictly only ever appended to the blob.
+Once appended they will never change and they will never move.
+
+If you know the byte offset in the blob for the start of the mmr data then you can check the number of mmr nodes currently in it by doing `(blobSize - mmrDataStart)/32`.
+
+## The peak stack and mmr data sizes are computable.
+
+See [Massif Blob Pre-Calculated Offsets](/developers/developer-patterns/massif-blob-offset-tables) to avoid needing to calculate these.
+Implementations of the O(log base 2 n) algorithms are provided in various languages.
+They all have very hardware-sympathetic implementations.
+
+## The massif height is constant for all blobs in a log *configuration*
+
+For massif height 14, the fixed size portion is `1048864` bytes.
+
+All massifs in a log are guaranteed to be the same *height*.
+If your log is re-configured having first been available at
+
+`https://app.datatrails.ai/verifiabledata/merklelogs/v1/mmrs/tenant/72dc8b10-dde5-43fe-99bc-16a37fd98c6a/0/`
+
+Then on re-configuration it will become available (without downtime) at:
+
+`https://app.datatrails.ai/verifiabledata/merklelogs/v1/mmrs/tenant/72dc8b10-dde5-43fe-99bc-16a37fd98c6a/1/`
+
+And the previous path will no longer receive any additions.
+
+{{< note >}}
+For the forseeable future (at least months) we don't anticipate needing to do this.
+{{< /note >}}
+
+## How to read a specific mmr node by its *mmrIndex*
+
+Find the smallest "Last Node" in [Massif Blob Pre-Calculated Offsets](/developers/developer-patterns/massif-blob-offset-tables) that is greater than your *mmrIndex* and use that row as your massif index
+
+Then taking massif index 0 (row 0) for example, and using the first mmrIndex for ease of example
+
+```python
+LOGSTART=1048864
+MMRINDEX=0
+curl -s \
+-H "Range: bytes=$(($LOGSTART+$MMRINDEX*32))-$(($LOGSTART+$MMRINDEX*32+31))" \
+-H "x-ms-blob-type: BlockBlob" \
+-H "x-ms-version: 2019-12-12" \
+https://jitavidfd1103b1099ab3aa.blob.core.windows.net/\
+merklelogs/v1/mmrs/tenant/73b06b4e-504e-4d31-9fd9-5e606f329b51/0/massifs/\
+0000000000000000.log  | od -An -tx1 | tr -d ' \n'
+
+a45e21c14ee5a0d12d4544524582b5feb074650e6bb2b31ed9a3aeffe4883099
+```
+
+The veracity demo tool can be used to confirm that
+
+```bash
+go run veracity/cmd/veracity/main.go -s jitavidfd1103b1099ab3aa \
```

Comment:
It'd be helpful to update the tool (and this example) to take a massif index too